Feature/add mapping link

zsloan opened the pull request:
This PR will add the option to get a shareable link for mapping results.

The way I'm planning on doing this will rely on the caching of the actual mapping tools themselves. While I would like to avoid re-running all the code for actually drawing the figure (etc), it isn't really practical due to how much of a mess the mapping display code is. The simplest way to do this seems to be to store the options as they exist at the beginning of run_mapping.py, because at that point the options can be JSON serialized. In display_mapping_results.py the options are no longer JSON serializable, so I'm not sure how to store them at that point without having to add a bunch of extra code that allows the dataset to be passed as either a string or object, etc.

zsloan committed:
Get and store JSON of mapping options in Redis for one hour whenever mapping is run

zsloan committed:
Add button for copying mapping results link

zsloan committed:
Add GET method to /run_mapping allowing it to pull the start_vars from Redis when an id is provided

BonfaceKilz commented:
I'll have a look at this later today.  Thanks!

-- 
(Life is like a pencil that will surely run out,
    but will leave the beautiful writing of life.)
(D4F09EB110177E03C28E2FE1F5BBAE1E0392253F
    (hkp://keys.openpgp.org))

zsloan commented:
> I'll have a look at this later today. Thanks!
> […](#)
> -- (Life is like a pencil that will surely run out, but will leave the beautiful writing of life.) (D4F09EB110177E03C28E2FE1F5BBAE1E0392253F (hkp://keys.openpgp.org))

It's not finished yet so no rush (though I might end up finishing it later tomorrow if it ends up being as straight-forward as I'm hoping it will be).

BonfaceKilz commented:
zsloan ***@***.***> aliandika:

>> I'll have a look at this later today. Thanks!
>> […](#)
>> -- (Life is like a pencil that will surely run out, but will leave the beautiful writing of life.) (D4F09EB110177E03C28E2FE1F5BBAE1E0392253F (hkp://keys.openpgp.org))
>
> It's not finished yet so no rush (though I might end up finishing it later tomorrow if it ends up being as straight-forward as I'm hoping it will be).

Sure!

-- 
(Life is like a pencil that will surely run out,
    but will leave the beautiful writing of life.)
(D4F09EB110177E03C28E2FE1F5BBAE1E0392253F
    (hkp://keys.openpgp.org))

BonfaceKilz commented:
zsloan ***@***.***> aliandika:

>> I'll have a look at this later today. Thanks!
>> […](#)
>> -- (Life is like a pencil that will surely run out, but will leave the beautiful writing of life.) (D4F09EB110177E03C28E2FE1F5BBAE1E0392253F (hkp://keys.openpgp.org))
>
> It's not finished yet so no rush (though I might end up finishing it later tomorrow if it ends up being as straight-forward as I'm hoping it will be).
>

Sure!
-- 
(Life is like a pencil that will surely run out,
    but will leave the beautiful writing of life.)
(D4F09EB110177E03C28E2FE1F5BBAE1E0392253F
    (hkp://keys.openpgp.org))

zsloan committed:
Add endpoint for caching mapping inputs

zsloan committed:
Add JS for caching mapping inputs + displaying the link

zsloan committed:
Add hash_of_inputs in display_mapping_results (so it can be displayed in the template)

zsloan committed:
Change Redis.hset to Redis.set so each caching can have a separate expiration

zsloan committed:
Move the code that caches mapping inputs to views.py so it can more easily include the temp_uuid (plus it's more consistent with the way caching has worked elsewhere)

Also changed the default time from two months to one month

zsloan committed:
Improve UI for mapping link

The link is now always displayed, but it is only saved (for longer than 1 hour) if the user clicks Copy and Share (which automatically copies the link)

The width of the input field (in ch) is multiplied by 0.95 because it will always be a bit shorter than the length of the link in ch (which multiplies its length by the width of the character 0). This is the best solution I could come up with for setting the width of the field.

zsloan commented:
Should have marked this as ready a while ago, since it seems to have been working fine when Rob's been using it from my branch.

robwwilliams commented:
Thanks Zach.

On Sat, Feb 11, 2023 at 19:33 zsloan ***@***.***> wrote:

> Should have marked this as ready a while ago, since it seems to have been
> working fine when Rob's been using it from my branch.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/763#issuecomment-1426916508>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC55V7HVZ5JIJK572CYQWMTWXA4XXANCNFSM6AAAAAAUAZNZMU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
Rob

Robert W Williams PhD
Department of Genetics, Genomics and Informatics
71 S Manassas St, Memphis TN 38163
University of Tennessee Health Science Center, TSRB 405
 Cell: 901 604 4752
EMAIL: ***@***.***
Zoom: https://tennesseehipaa.zoom.us/j/9492269658
----------------
